File: javascript/prime-factors/prime-factors.js
class PrimeFactors {
  
    for(num) {
        let oa = num;
        let i = 2;
        let arr = [];
        while (oa != arr.reduce((x,y) => x * y,1)) {
            if (num % i === 0) {
                arr.push(i);
                num /= i;
            } else {
                i++;
            }
        }
        return arr;
    }
 
}

var primeFactors = new PrimeFactors();
   
module.exports = primeFactors;

File: ruby/prime-factors/prime_factors.rb
require 'prime'

class PrimeFactors
  def self.for(num)
    oa = num
    i = 0
    arr = []
    primes = Prime.take(71_000)

    return [] if num <= 1

    loop do
      break if oa == arr.reduce(&:*)
      if (num % primes[i]).zero?
        arr << primes[i]
        num /= primes[i]
      else
        i += 1
      end
    end
    arr
  end
end
